connect-inject/health_check_resource.go

```diff
@@ -232,6 +234,13 @@ func (h *HealthCheckResource) getServiceCheck(client *api.Client, healthCheckID
 // ready state of the pod along with the reason message which will be passed into the Notes
 // field of the Consul health check.
 func (h *HealthCheckResource) getReadyStatusAndReason(pod *corev1.Pod) (string, string, error) {
+	// A pod might be pending if the init containers have run but the non-init
+	// containers haven't been started. In this case we set a failing health
+	// check so the pod doesn't receive traffic before it's ready.
+	if pod.Status.Phase == corev1.PodPending {
+		return api.HealthCritical, podPendingReasonMsg, nil
+	}
+
```

Comment:
Yeah good question, my idea was that it would be better to say `Pod is pending` than `containers with unready status: [static-client consul-connect-envoy-sidecar consul-connect-lifecycle-sidecar]` because I thought that would be more understandable. For me when I see the latter I would wonder if the readiness checks were failing.

Here's what the pod status looks like when the init container has terminated but it's still pending:

``json

```

Reply:
Thanks for the explanation! Yeah that makes sense, `Pod is pending` sounds like something you would just have to wait for, while `containers with unready status: [static-client consul-connect-envoy-sidecar consul-connect-lifecycle-sidecar]` sounds like an error you'd want to fix.